tests/app/notifications/rest/test_send_notification.py

```diff
@@ -944,3 +941,65 @@ def test_should_delete_sms_notification_and_return_error_if_sqs_fails(notify_api
         assert response.status_code == 500
         assert not notifications_dao.get_notification_by_id(fake_uuid)
         assert not NotificationHistory.query.get(fake_uuid)
+
+
+@pytest.mark.parametrize('to_email', [
+    'simulate-delivered@notifications.service.gov.uk',
+    'simulate-permanent-failure@notifications.service.gov.uk',
+    'simulate-temporary-failure@notifications.service.gov.uk'
+])
+def test_should_not_persist_notification_or_send_email_if_simulated_email(
+        client,
+        to_email,
+        sample_email_template,
+        fake_uuid,
+        mocker):
+    mocker.patch('app.notifications.rest.create_uuid', return_value=fake_uuid)
+    apply_async = mocker.patch('app.celery.provider_tasks.send_email_to_provider.apply_async')
+
+    data = {
+        'to': to_email,
+        'template': sample_email_template.id
+    }
+
+    auth_header = create_authorization_header(service_id=sample_email_template.service_id)
+
+    response = client.post(
+        path='/notifications/email',
+        data=json.dumps(data),
+        headers=[('Content-Type', 'application/json'), auth_header])
+
+    assert response.status_code == 201
+    assert not notifications_dao.get_notification_by_id(fake_uuid)
+    apply_async.assert_not_called()
```

Comment:
You could assert Notification.query.count() == 0, then avoid the first mocker for create_uuid, shave a couple of lines of code.